### ISSUE TITLE
[Suggestion] Interfacing Courseplay

### ISSUE THREAD

**Issue by pvajko** [Mogli12/AIVehicleExtension#102]
I'm one of the Courseplay devs working mostly on fieldwork course generation. As you probably know Courseplay is not really good at corners, leaving fruit on the field (see for example https://github.com/Courseplay/courseplay/issues/2242).

We can certainly improve that with reasonable effort but we have AIVehicleExtension here which does a really great job at those corners so I was wondering if we could make these two mods work together. 

The purpose of creating this issue here is to discuss with @Mogli12 (and @pops64) the feasibility of such an interface. 

Would it be possible that when Courseplay, driving for example a combine on a fieldwork course, detects a situation which can better be handled by AIVE (like a sharp headland corner), would pass on control of the combine to AIVE? And when AIVE finishes the maneuver, gives control back to Courseplay?

**Comment by alpha117** [comment]
That would be Fantastic, if you guys could make this happen

**Comment by pvajko** [comment]
I think it's doable in Courseplay, not sure about AIVE, a problem I see is that when AIVE is turned on, it scans the field which takes a while. That could probably changed so it scans only an area big enough for the turn?

**Comment by Mogli12** [comment]
AIVE and courseplay work completely different. AIVE does not follow a course. It calculates the best steering angle based on the current fruit situations.

**Comment by Dzi4d3k** [comment]
@Mogli12 I think that's exactly why @pvajko asked about 'cooperation' between AIVE and Courseplay in this issue, as AIVE is better in handling tough field situation. I believe that question is about possibility to create, let say, 'set of variables', which will be used to alternate control of artificial worker between between being controlled by AIVE or Courseplay.

**Comment by pvajko** [comment]
I think I know how Courseplay works as I implemented a new fieldwork course generator for it. And I know it is not the best at handling corners without missing fruit.

The point I tried to make: let AIVE make the corners by calculating the best steering angle and then return the control of the vehicle to Courseplay. CP would then give control over the vehicle to AIVE again when it encounters a situation that AIVE can handle better.
